### PR TITLE
Feature/22322 admin familiarisation pins

### PR DESCRIPTION
**Pull request by danwilliams**

### CODE REVIEW

File: admin/assets/javascripts/global-scripts.js [hunk 1]
```diff
@@ -642,4 +642,28 @@ $(function () {
       $('#' + $(this).context.id).text('(' + totalPupils + ' pupil' + (totalPupils > 1 ? 's' : '') + ')')
     })
   }
+
+  /**
+   * Filtering pupils by name.
+   */
+  if ($('.filter-name').length > 0) {
```
Comment (GeorgeChatzigiannis) [inline admin/assets/javascripts/global-scripts.js]: @danwilliams @jon-shipley @GuyHarwood I am worried global scripts is becoming quite large and not easily maintainable. Would it be a right time to start moving to separate files depending on purpose?

Comment (GuyHarwood) [reply admin/assets/javascripts/global-scripts.js]: @GeorgeChatzigiannis yes, absolutely

File: admin/data/sql/migrations/106.do.add-familiarisation-pin.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+ALTER TABLE [mtc_admin].[pupil] ADD familiarisation_pin NVARCHAR(12) NULL
```
Comment (GuyHarwood) [inline admin/data/sql/migrations/106.do.add-familiarisation-pin.sql]: we definitely don't want 2 pin fields.  the bit field distinguishes this.  We went through this in fridays discussion (you may have joined after it was discussed?)
also, we are moving to a new table `checkFormAllocation`.  I mentioned to the BAs that this card was supposed to just cover the UI, and not actually do the plumbing, as that will be done by an azure function, triggered by a queue message.

File: admin/controllers/questions.js [hunk 3]
```diff
@@ -53,7 +53,7 @@ const getQuestions = async (req, res) => {
         await storeLogonEvent(data.pupil.id, schoolPin, pupilPin, false, 500, errorMessage)
         return apiResponse.serverError(res)
       }
-      await checkStartService.prepareCheck([data.pupil.id], data.school.dfeNumber, data.school.id)
+      await checkStartService.prepareCheck([data.pupil.id], data.school.dfeNumber, data.school.id, 'live')
```
Comment (GuyHarwood) [inline admin/controllers/questions.js]: this should always be familiarisation mode for the test account.

File: admin/data/sql/migrations/106.do.add-familiarisation-pin.sql [hunk 4]
```diff
@@ -0,0 +1,3 @@
...
+ALTER TABLE [mtc_admin].[pupil] ADD familiarisation_pinExpiresAt [datetimeoffset](3) NULL
```
Comment (GuyHarwood) [inline admin/data/sql/migrations/106.do.add-familiarisation-pin.sql]: same as above.

File: admin/data/sql/migrations/106.do.add-familiarisation-pin.sql (hunk 2 above)
Comment (danwilliams) [reply admin/data/sql/migrations/106.do.add-familiarisation-pin.sql]: Ah okay, I was under the impression that the code from e.g. this feature would be moved to the Azure function after it was working. Should I remove the "functionality" for pin generation and address it after the Azure function / queue message is implemented to do that?

Comment (GuyHarwood) [reply admin/data/sql/migrations/106.do.add-familiarisation-pin.sql]: you can stub the familiarisation mode out for now.  Jon has already picked up the ticket that covers that part of it i believe.

The UI side of things looks good, so we can just wire it up when Jons done the other ticket 😎

**Comment by jon-shipley** [comment]
@danwilliams  Please resolve the conflicts.

**Pull request merged**